src/main/java/tracko/logic/commands/order/EditOrderCommand.java

```diff
@@ -181,21 +211,63 @@ public Optional<Address> getAddress() {
             return Optional.ofNullable(address);
         }
 
-        /**
-         * Sets {@code tags} to this object's {@code tags}.
-         * A defensive copy of {@code tags} is used internally.
-         */
-        public void setTags(Set<Tag> tags) {
-            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        public Optional<List<ItemQuantityPair>> getItemList() {
+            return Optional.ofNullable(itemList);
+        }
+
+        public void setItemList(List<ItemQuantityPair> itemList) {
+            this.itemList = itemList;
         }
 
         /**
-         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
-         * if modification is attempted.
-         * Returns {@code Optional#empty()} if {@code tags} is null.
+         * Updates the item list. This method checks whether the item that is going to be updated is
+         * in the customer's ordered items list; if it is, then it will check whether the values are different
+         * from the original or 0. If the value is zero, the item will be removed from the order list, else
+         * its quantity will be updated. If the item is not in the item list, then the item will be added to
+         * the customer's ordered items list.
+         *
+         * @param orderToEdit The order whose ordered items list is going to be edited.
+         * @param itemToEdit The item that is going to be added, removed or have its quantity edited.
          */
-        public Optional<Set<Tag>> getTags() {
-            return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
+        public void updateItemList(Order orderToEdit, ItemQuantityPair itemToEdit) throws CommandException {
+            List<ItemQuantityPair> orderedItems = orderToEdit.getItemList();
+            boolean hasItemBeenUpdated = false;
+            for (int i = 0; i < orderedItems.size(); i++) {
+                ItemQuantityPair itemInList = orderedItems.get(i);
+
+                // Check whether the item is in the customer's order list.
+                boolean isItemInList = itemInList.getItemName().equalsIgnoreCase(itemToEdit.getItemName());
+                boolean isUpdatedQuantityZero = itemToEdit.getQuantityValue() == 0;
```

Comment:
I think you can also move this boolean statement out of the for loop if I'm not wrong